sentry/src/db/campaign.rs

```diff
@@ -485,4 +582,210 @@ mod test {
             );
         }
     }
+
+    async fn insert_multiple_campaigns_with_different_channels(pool: &DbPool) {
+        let campaign = DUMMY_CAMPAIGN.clone();
+        let mut channel_with_different_leader = DUMMY_CAMPAIGN.channel;
+        channel_with_different_leader.leader = IDS["user"];
+
+        let _channel = insert_channel(&pool, DUMMY_CAMPAIGN.channel)
+            .await
+            .expect("Should insert");
+
+        let _channel = insert_channel(&pool, channel_with_different_leader)
+            .await
+            .expect("Should insert");
+
+        let mut campaign_2 = DUMMY_CAMPAIGN.clone();
+        campaign_2.id = CampaignId::new();
+
+        // campaign with a different creator
+        let mut campaign_3 = DUMMY_CAMPAIGN.clone();
+        campaign_3.id = CampaignId::new();
+        campaign_3.creator = ADDRESSES["tester"];
+
+        let mut campaign_4 = DUMMY_CAMPAIGN.clone();
+        campaign_4.id = CampaignId::new();
+
+        campaign_4.channel = channel_with_different_leader;
+        campaign_4.validators = Validators::new((
+            DUMMY_VALIDATOR_DIFFERENT_LEADER.clone(),
+            DUMMY_VALIDATOR_FOLLOWER.clone(),
+        ));
+
+        let is_inserted = insert_campaign(&pool, &campaign)
+            .await
+            .expect("Should succeed");
+
+        assert!(is_inserted);
+
+        let is_inserted = insert_campaign(&pool, &campaign_2)
+            .await
+            .expect("Should succeed");
+
+        assert!(is_inserted);
+
+        let is_inserted = insert_campaign(&pool, &campaign_3)
+            .await
+            .expect("Should succeed");
+
+        assert!(is_inserted);
+
+        let is_inserted = insert_campaign(&pool, &campaign_4)
+            .await
+            .expect("Should succeed");
+
+        assert!(is_inserted);
+    }
+
+    #[tokio::test]
+    async fn it_lists_campaigns_properly() {
+        let database = DATABASE_POOL.get().await.expect("Should get a DB pool");
+
+        setup_test_migrations(database.pool.clone())
+            .await
+            .expect("Migrations should succeed");
+
+        insert_multiple_campaigns_with_different_channels(&database.pool).await;
```

Comment:
this might become a problem when a test fails since Rust does not show you properly where it has failed when calling functions with assert-ion checks.